### ISSUE TITLE
Bug: cockpit-tls facing SELinux issues

### ISSUE THREAD

**Issue by daniel-wtd** [cockpit-project/cockpit#13275]
Cockpit version: cockpit-ws-208-1.fc31

OS: Fedora 31 (IOT)

After a fresh installation of cockpit and enabeling cockpit.socket, the website on port 9090 does not load due to selinux errors.

Steps to reproduce:

1. Install Fedora IOT
2. `rpm-ostree install cockpit-ws`
3. `systemctl reboot*
4. `systemctl enable --now cockpit.socket`
5. firewall-cmd --add-service cockpit
6. point browser to <ip>:9090 -> not working
7. `setenforce permissive`
8. point browser to <ip>:9090 -> working

```
Dec 10 22:00:44 localhost.localdomain audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=cockpit-ws>
Dec 10 22:00:44 localhost.localdomain cockpit-tls[2473]: cockpit-tls: gnutls_handshake failed: A TLS fatal alert has been received.
Dec 10 22:01:00 localhost.localdomain audit[2473]: AVC avc:  denied  { connectto } for  pid=2473 comm="cockpit-tls" path="/run/cockpit/wsinstance/https-factor>
Dec 10 22:01:00 localhost.localdomain cockpit-tls[2473]: cockpit-tls: connect(/run/cockpit/wsinstance/https-factory.sock) failed: Permission denied
```

**Comment by martinpitt** [comment]
Our `%post` has a hack for this, but this should be finally fixed for good in https://bodhi.fedoraproject.org/updates/FEDORA-2019-fefda9dd5e , i. e. in selinux-policy ≥ 3.14.4-41. That made it to Fedora 31 a few hours ago. Could you please re-test with that?

**Comment by daniel-wtd** [comment]
Maybe it needs some time to be present in the repositories. For now: no new update available in F31 and F31-IOT.

**Comment by martinpitt** [comment]
We dropped the `%post` hack in 209, and Fedora 30 and 31 finally have a current selinux-policy that works.

**Comment by daniel-wtd** [comment]
Awesome, thanks a lot :)

**Comment by jackxujh** [comment]
I had the same Permission Denied error (from systemctl log) with a fresh Fedora 31 Workstation. However, after setting `setenforce 0` then `setenforce 1`, I am able to access Cockpit from a LAN client. 

However, I could not understand why I would have `Permission Denied` in the first place. Does this have something to do with the fix in this discussion? Thanks!

**Comment by jackxujh** [comment]
Sorry, I was wrong, it turned out I have to keep `setenforce permissive` or `setenforce 0` to be able to use Cockpit.

**Comment by martinpitt** [comment]
@jackxujh : Can you please open a fresh issue (or even better, bugzilla), with the details? Please run `sudo journalctl -f`, then try to log into Cockpit, and paste the journal output. This should include the SELinux policy violation details.

**Comment by jackxujh** [comment]
@martinpitt thanks for your response! I was able to get my problem fixed with the operations you suggested in #13183. So basically I reinstalled Cockpit, and other packages. I am not sure though if I could reproduce this error at this moment, since I don't have more computers not having Cockpit installed.

**Comment by martinpitt** [comment]
@jackxujh : `journalctl` is not fixing anything, it just shows the system logs. So this somehow fixed itself earlier already..
